src/elements/Asset.php

```diff
@@ -1140,11 +1140,12 @@ public function getTitleTranslationKey(): string
      */
     public function getFieldLayout()
     {
-        if (($fieldLayout = parent::getFieldLayout()) !== null) {
```

Comment:
I take it that `$fieldLayout` was getting set here? Maybe because something was calling `setFieldLayout()` earlier?

Reply:
`setFieldLayout()` is not actually called here. The layout is determined based on the `defaultFieldLayoutId` body param. 

The order of events is as follows:
- `Asset->getEditorHtml()` starts with `fieldLayoutId = null`, which means it then uses the `defaultFieldLayoutId` body param (https://github.com/craftcms/cms/blob/v3/src/elements/Asset.php#L1821-L1828)
- then `Element->getEditorHtml()` checks if the element has a field layout, 
- `Element::hasFieldLayout()` calls `Asset::getFieldLayout()`  
- which calls `Element::getFieldLayout()`, which gets the layout by `fieldLayoutId`, which at this point is set to the `defaultFieldLayoutId` body param 
- and so the default layout is returned

Reply:
So maybe we just need to get rid of `defaultFieldLayoutId`? I’m guessing that was added before we had the modern element editor.

Can you test removing that, and then make sure that the proper field layout is still getting shown for freshly-uploaded assets, when they are able to be uploaded directly to a real (non-temp) volume?

Reply:
Yup, you’re right. `defaultFieldLayoutId` is no longer needed. I have:
- an “Uploads” volume on Local FS with the field layout containing a “Number” field
- an “S3” volume on S3 FA with the field layout containing a “Plain Text” field
- Temporary uploads location set to a temp folder (so no field layout)
- autosaving drafts is disabled


https://user-images.githubusercontent.com/4500340/223683291-d8b1959e-0b05-4238-82ca-265419a91ab2.mov

Also tested with autosaving drafts enabled, in which case the file gets sent to the default uploads location (S3 in my case), and when you double-click it, it shows the S3 field layout.